aptos-move/aptos-vm/src/verifier/randomness.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::move_vm_ext::{AptosMoveResolver, SessionExt};
+use aptos_types::transaction::EntryFunction;
+use move_binary_format::errors::VMResult;
+
+/// Returns true if function has an attribute that it uses randomness.
+pub(crate) fn has_randomness_attribute(
+    resolver: &impl AptosMoveResolver,
+    session: &mut SessionExt,
+    entry_fn: &EntryFunction,
+) -> VMResult<bool> {
+    let module = session
+        .get_move_vm()
+        .load_module(entry_fn.module(), resolver)?;
```

Comment:
it's a bit weird that this would require resolver, isn't the session already holding a module_store?

Reply:
Yes, I was trying to re-use it from the session, but I think the problem is that we store a trait object `dyn AptosMoveResolver` in session, which rust doesn't allow to upcast to `MoveResolver` (unless I add `AsBaseMoveResolver` trait or similar), so I kept it this way. Will give it another try!